src/hammer-vlsi/hammer_utils/__init__.py

```diff
@@ -254,11 +255,23 @@ def is_union(t: Any) -> bool:
     def compare_types_internal(a: Any, b: Any) -> bool:
         """
         Comparing types is also complicated.
-        Particularly when you have native Python times and mypy types floating around at once.
+        Particularly when you have native Python types and mypy types floating around at once.
         WARNING: this method is in no way complete/exhaustive
         """
         import typing
-        if a == dict and b == typing.Dict:
+
+        if isinstance(a, str) and isinstance(b, str):
+            # Two strings (e.g. if both are stringly-typed mypy annotations).
+            # Just check if they are identical.
+            return a == b
+        elif isinstance(a, str) and isinstance(b, type):
+            # A string and a type.
+            # Likely the first is a mypy stringly-typed annotation and the
```

Comment:
What happens in the "unlikely" case?

Reply:
What I mean is that if someone writes nonsensical in the stringly-typed annotation (or is abusing the annotation field for other stuff), it will obviously fail as expected.

```python
def my_func(x: "this is a totally nonsensical type annotation", y: int) -> bool: ...
```